### PR TITLE
Better document how to configure the dependency resolvers

### PR DESCRIPTION
**Pull request by shs96c**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1146,6 +1147,46 @@ Or, to publish to a GCP Artifact Registry:
...
+### Configuring Maven
+
+A Maven-backed resolver can be used by using setting the `resolver`
+attribute to `maven`. This resolver requires the use of a lock file. For
+bootstrapping purposes, this file may simply be an empty file. When using
+the maven-backed resolver, the following environment variables are honoured:
+
+| Environment variable | Meaning                                                                                                      |
+|----------------------|--------------------------------------------------------------------------------------------------------------|
+| `RJE_ASSUME_PRESENT` | Prevents the resolver from checking remote caches to see if an dependency is present, and just assumes it is |
```
Comment (jin) [inline README.md:1178]:
```suggestion
| `RJE_ASSUME_PRESENT` | Prevents the resolver from checking remote caches to see if a dependency is present, and just assumes it is |
```

File: README.md [hunk 2]
```diff
@@ -1146,6 +1147,46 @@ Or, to publish to a GCP Artifact Registry:
...
+| `RJE_MAX_THREADS`    | Integer giving the maximum number of threads to use for downloads                                            |
```
Comment (jin) [inline README.md:1179]: Also mention the default?

File: README.md [hunk 3]
```diff
@@ -1146,6 +1147,46 @@ Or, to publish to a GCP Artifact Registry:
...
+| `RJE_UNSAFE_CACHE`   | When set to `1` will use your `$HOME/.m2/repository` directory to speed up dependency resolution             |
+| `RJE_VERBOSE`        | When set to `1` extra diagnostic logging will be sent to `stderr`                                            |
```
Comment (jin) [inline README.md:1181]: nit: this could be factored in a common option.

**Review by jin — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (shs96c) [reply README.md:1179]: Done

File: README.md (hunk 3 above)
Comment (shs96c) [reply README.md:1181]: Done

**Pull request merged**